store/package-store/src/storeController/prune.ts

```diff
@@ -24,7 +24,7 @@ export async function prune ({ cacheDir, storeDir }: PruneOptions, removeAlienFi
   globalInfo('Removed all cached metadata files')
   const pkgIndexFiles = [] as string[]
   const removedHashes = new Set<string>()
-  const dirs = (await fs.readdir(cafsDir, { withFileTypes: true }))
+  const dirs = (await fs.readdir(cafsDir, { withFileTypes: true }).catch(() => []))
```

Comment:
now you ignore any error not just ENOENT

Reply:
@zkochan changed in https://github.com/pnpm/pnpm/pull/8555/commits/82da769ee81ac7af2a1a931a0bc0bdb1c8e20dd9 to only ignore `ENOENT` errors.